go/kbfs/libkbfs/kbfs_ops.go

```diff
@@ -300,6 +300,43 @@ func (fs *KBFSOpsStandard) GetFavoritesAll(ctx context.Context) (
 		return keybase1.FavoritesResult{}, err
 	}
 
+	// Add the sync config mode to each favorite.
+	tlfIDs := fs.config.GetAllSyncedTlfs()
+	for _, id := range tlfIDs {
+		config, err := fs.GetSyncConfig(ctx, id)
+		if err != nil {
+			return keybase1.FavoritesResult{}, err
+		}
+
+		if config.Mode == keybase1.FolderSyncMode_DISABLED {
+			panic(fmt.Sprintf(
+				"Folder %s has sync unexpectedly disabled", id))
+		}
+
+		fb := data.FolderBranch{Tlf: id, Branch: data.MasterBranch}
+		_, h, err := fs.GetRootNodeMetadata(ctx, fb)
+		if err != nil {
+			return keybase1.FavoritesResult{}, err
+		}
+
+		name := string(h.GetCanonicalName())
+		for i, fav := range favs.FavoriteFolders {
```

Comment:
Instead of doing a linear search every time, can you create a set/map one time above this loop and just index into it? Some people have lots of favorites and this could hurt performance.

Reply:
It would only hurt performance if they also had a lot of synced TLFs, and your suggestion would hurt performance a little if they only had one (overhead for the map creation) and makes the code more complex (I'll have to define a new type for the map key, I think).  I don't know what the right answer is performance-wise.  I'll give it a try and see how it looks.

Reply:
I suspect this would be a win with like 3+ synced TLFs.

Reply:
Cool I pushed a new commit that does this optimization.  Let me know if it looks preferable to you.